### ISSUE TITLE
[java] Improve quality of AccessorClassGeneration

### ISSUE THREAD

**Issue by vanniktech** [pmd/pmd#291]
**Rule Set:**
AccessorClassGeneration (java-design)

**Description:**
I've found a way of not triggering the AccessorClassGeneration even though it should as the synthetic methods and classes are generated.

**Code Sample demonstrating the issue:**

```java
public class Example extends View {
  private Example(final Context context) {
    super(context);
  }

  public static void create() {
    new Handler().post(new Runnable() {
      @Override public void run() {
        Example example = new Example(null); // Should be marked since the constructor is private.
      }
    });
  }
}
```

**Running PMD 5.5.4 through:** *[Gradle]*

**Comment by jsotuyod** [comment]
@vanniktech thanks for the report!

At first sight this seems like a duplicate of #274, which was fixed in #276 (to be released in PMD 5.5.5 / 5.6.0).

I'll double check just in case and let you know.

**Comment by jsotuyod** [comment]
Ok, this is interesting. The first 2 cases are duplicates and are actually fixed on the repository. The other case (calling a private constructor) is not the scope of `AccessorMethodGeneration` but of `AccessorClassGeneration`.

I'll keep this report, but reword it appropriately.

**Comment by vanniktech** [comment]
Oh okay cool. Thanks for sorting it out. 👍
